### DISCUSSION TITLE
Icons in ListView slow to update

### DISCUSSION THREAD

**Discussion by HakanL** [unoplatform/uno#14047] [Q&A]
I have a strange issue, which I'm not sure if it's something I'm doing, or a bug. I have a page with a ListView, which then has a WrapPanel from the community toolkit. I'm displaying Toolkit Cards in it with an icon, two lines of text. It looks like this, works great:
![image](https://github.com/unoplatform/uno/assets/407941/98fae925-dd2f-4944-9829-b6dfadc6c3ba)

If I click on one of the cards, I'm navigating to a sub menu, which is essentially the same view, but with a new instance of a view model (that has a new set of menu items, in a `ObserverableCollection`). The issue is that for a split second (which is more obvious on a RPi that is slower to process) the icons from the first list of menu items are used on the new list. It looks like this:
![image](https://github.com/unoplatform/uno/assets/407941/b5b8f146-7c92-419b-97e8-c540262af464)

But the next "frame" shows the correct icons:
![image](https://github.com/unoplatform/uno/assets/407941/ff2f4f3a-179b-4b49-9b0d-87aac0e0ebe6)

I can't figure out why this happens. I've traced the code, I get my new instance of the VM, I set the list of menu items, it never has the old icons with the new menu items. It's like the icons are cached or something. Is there a workaround where I can pause screen refresh until everything is updated, or another way to solve it? It looks strange when the icons changed after the menu is updated.

**Comment by jeromelaban** [comment]
Thanks for the report! 

This is likely caused by template reuse, where unbound images are not getting the contents reset. You may be able to work around it by explicitly calling `.Clear()` on the list, then adding the new items. If that's what you do already, it may be caused by something different, like template pooling ([which you can disable](https://github.com/unoplatform/uno/blob/2af7b1a6953cc501dd6b16c07c4510fd8a937828/src/Uno.UI/UI/Xaml/FrameworkTemplatePool.cs#L116) to see if it helps).

Let us know what you find, and we'll likely open an issue to track this.

**Reply by HakanL** [reply]
Yes, `.Clear()` is called (indirectly by using the ReplaceWith extension, but I've already tried to explicitly call `.Clear()` and it makes no difference). When the view renders then the items are already populated, it may be possible to do a notify of property changed after Clear, but I don't necessarily want to display an empty list and then have it populate.
I tried to disable Template Pooling which at least prevents the wrong icon from being displayed, however now the menu items are rendered without any icons for 2 frames (60 fps using screen recorder) and then the correct icon is displayed. I'm not sure that's better, because now the text starts out left aligned, then it's pushed to the right when the icons are loaded. But it may point to where the issue may be?
![image](https://github.com/unoplatform/uno/assets/407941/b576fe3c-3fc4-459a-88bf-5ae9b95ba668)

**Reply by HakanL** [reply]
@jeromelaban Any other suggestions on how I can workaround this issue, it doesn't look good when the screen changes. Is it possible to have some blank icon while the correct ones are loading? So at least the text wouldn't move around? Or set a fixed icon size so the space is reserved?
